test/extended/dns/dns_libraries.go

```diff
@@ -0,0 +1,215 @@
+package dns
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+
+	buildv1 "github.com/openshift/api/build/v1"
+	configv1 "github.com/openshift/api/config/v1"
+	v1 "github.com/openshift/api/image/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	exutil "github.com/openshift/origin/test/extended/util"
+
+	kapiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	watchtools "k8s.io/client-go/tools/watch"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+// checkForPodLogFailures goes through the pod logs and determines if there was a failure
+// by looking for "fail" keyword in the logs.
+func checkForPodLogFailures(f *e2e.Framework, pod *kapiv1.Pod) {
+	g.By("creating the pod and waiting for success status")
+	podClient := f.ClientSet.CoreV1().Pods(f.Namespace.Name)
+	updated, err := podClient.Create(context.Background(), pod, metav1.CreateOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	w, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Watch(context.Background(), metav1.SingleObject(
+		metav1.ObjectMeta{
+			Name:            pod.Name,
+			ResourceVersion: updated.ResourceVersion,
+		},
+	))
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	ctx, cancel := context.WithTimeout(context.Background(), e2e.PodStartTimeout)
+	defer cancel()
+	if _, err = watchtools.UntilWithoutRetry(ctx, w, PodSucceeded); err != nil {
+		e2e.Failf("Failed: %v", err)
+	}
+
+	g.By("retrieving the pod logs")
+	r, err := podClient.GetLogs(pod.Name, &kapiv1.PodLogOptions{Container: "querier"}).Stream(context.Background())
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	scan := bufio.NewScanner(r)
+	for scan.Scan() {
+		line := scan.Text()
+		if strings.Contains(line, "fail") {
+			e2e.Failf("DNS resolution failed: %s", line)
+		}
+	}
+}
+
+var _ = g.Describe("[sig-network-edge] DNS lookup", func() {
+	f := e2e.NewDefaultFramework("dns-libraries")
+	oc := exutil.NewCLI("dns-libraries")
+	nodeSelector := make(map[string]string)
+
+	g.BeforeEach(func() {
+		controlPlaneTopology, err := exutil.GetControlPlaneTopology(oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if *controlPlaneTopology != configv1.ExternalTopologyMode {
+			nodeSelector["node-role.kubernetes.io/master"] = ""
+		}
+	})
+
+	// Create pods using the image built from the Dockerfile located in testdata/dns/go-dns-resolver.
+	// These pods perform a DNS query to verify different versions of Go's DNS resolver function correctly
+	// with OpenShift DNS.
+	g.It("using Go's DNS resolver", func() {
+		configClient, err := configclient.NewForConfig(f.ClientConfig())
+		o.Expect(err).NotTo(o.HaveOccurred())
+		infra, err := configClient.ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
```

Comment:
You may want to skip MicroShift, as it does not support `config.openshift.io` api group. An example on how to do it: https://github.com/openshift/origin/blob/e6b779017d5e8ebbdbfb84ec0105f0033d6eb1b9/test/extended/operators/certs.go#L95-L99

Reply:
Thanks - I didn't notice the MicroShift failure. We just need the infra config to figure out if it's baremetal, so I tried skipping the check for microshift to see if the test will pass on microshift. As long as it can pull external images, it should be good.

Reply:
Unfortunately it looks like this test needs skipping, the build APIs are not present in MicroShift.